crates/chia-consensus/src/multiprocess_validation.rs

```diff
@@ -0,0 +1,509 @@
+
+// currently in mempool_manager.py
+// called in full_node.py when adding a transaction
+pub fn pre_validate_spendbundle(
+    new_spend: &SpendBundle,
+    max_cost: u64,
+    constants: &ConsensusConstants,
+    peak_height: u32,
+    cache: Arc<Mutex<BlsCache>>,
+) -> Result<OwnedSpendBundleConditions, ErrorCode> {
+    if new_spend.coin_spends.is_empty() {
+        Err(ErrorCode::InvalidSpendBundle)
+    } else {
+        let (result, _added, _duration) =
+            validate_clvm_and_signature(new_spend, max_cost, constants, peak_height, cache)?;
+        Ok(result)
+    }
+}
+
+// currently in mempool_manager.py
+// called in threads from pre_validate_spend_bundle()
+// returns (error, cached_results, new_cache_entries, duration)
+pub fn validate_clvm_and_signature(
+    spend_bundle: &SpendBundle,
+    max_cost: u64,
+    constants: &ConsensusConstants,
+    height: u32,
+    cache: Arc<Mutex<BlsCache>>,
+) -> Result<
+    (
+        OwnedSpendBundleConditions,
+        HashMap<[u8; 32], GTElement>,
+        Duration,
+    ),
+    ErrorCode,
+> {
+    let start_time = Instant::now();
+    let npcresult = get_name_puzzle_conditions(spend_bundle, max_cost, true, height, constants)
+        .map_err(|e| e.1)?;
+    let iter = npcresult.spends.iter().flat_map(|spend| {
+        // let spend_clone = spend.clone();
+        let condition_items_pairs = [
+            (AGG_SIG_PARENT, &spend.agg_sig_parent),
+            (AGG_SIG_PUZZLE, &spend.agg_sig_puzzle),
+            (AGG_SIG_AMOUNT, &spend.agg_sig_amount),
+            (AGG_SIG_PUZZLE_AMOUNT, &spend.agg_sig_puzzle_amount),
+            (AGG_SIG_PARENT_AMOUNT, &spend.agg_sig_parent_amount),
+            (AGG_SIG_PARENT_PUZZLE, &spend.agg_sig_parent_puzzle),
+            (AGG_SIG_ME, &spend.agg_sig_me),
+        ];
+        condition_items_pairs
+            .iter()
+            .flat_map(|(condition, items)| {
+                let spend = spend.clone();
+                items.iter().map(move |(pk, msg)| {
+                    (
+                        pk,
+                        make_aggsig_final_message(*condition, msg.as_slice(), &spend, constants),
+                    )
+                })
+            })
+            .collect::<Vec<_>>()
+    });
+    let unsafe_items = npcresult
+        .agg_sig_unsafe
+        .iter()
+        .map(|(pk, msg)| (pk, msg.as_slice().to_vec()));
+    let iter = iter.chain(unsafe_items);
+    // Verify aggregated signature
+    let (result, added) = cache
+        .lock()
+        .unwrap()
+        .aggregate_verify(iter, &spend_bundle.aggregated_signature);
+    if !result {
+        return Err(ErrorCode::InvalidSpendBundle);
+    }
+    Ok((npcresult, added, start_time.elapsed()))
+}
+
+pub fn get_flags_for_height_and_constants(height: u32, constants: &ConsensusConstants) -> u32 {
+    let mut flags: u32 = 0;
+    if height >= constants.soft_fork2_height {
+        flags |= NO_RELATIVE_CONDITIONS_ON_EPHEMERAL;
+    }
```

Comment:
this is in the pipeline of being removed as well. https://github.com/Chia-Network/chia-blockchain/pull/18135
In fact, in the latest version of `chia_rs`, this constant has been removed.